src/dafny/DafnyCore.md

```diff
@@ -0,0 +1,51 @@
+
+# Guide to the Dafny Core Library
+
+The Dafny programming language and tool set includes
+libraries that provide functionality in various domains. This document describes 
+the _Dafny Core Library_. Dafny programs are composed of modules; to ensure
```

Comment:
Note that this PR isn't intending to relocate code, but all the docs are still sitting under `src/dafny`, which doesn't yet contain any code. You'll have to move them to sit beside their respective Dafny files to make sense.

Given that, I don't actually think you want to talk about the "Dafny Core Library" yet because that only makes sense when referring to `src/dafny` in #92 instead. I think I lean towards just deleting this whole file from this PR and only adding it in #92.

Reply:
The intention was to reorganize the code and document the reorganization, rather than write documentation only to have to change it. Also, this file is the starting point for the other material. Without it, users won't have a way to get to it to comment.

And yes, the files are in these subdirectories so they sit next to their code (as you requested before). It does not hurt to have them there now with empty direcotries.